psh/test-cat.py

```diff
@@ -27,8 +25,61 @@ def assert_cat_h(p):
     psh.assert_cmd(p, 'cat -h', expected=help, result='success')
 
 
+def assert_cat_err(p):
+    fname = 'nonexistentFile'
+
+    statement = f'cat: {fname}: ENOENT'
+    psh.assert_cmd(p, f'cat {fname}', expected=statement, result='fail')
+
+    statement = rf'cat: {ROOT_TEST_DIR}: EISDIR'
+    psh.assert_cmd(p, f'cat {ROOT_TEST_DIR}', expected=statement, result='fail')
```

Comment:
We want to leave it flexible (for example on linux the behavior is a little but different: `cat: nonexistent: No such file or directory` - maybe we will change the approach in the future). However checking cat on directory is a good idea and we can leave it.